pkg/csi/service/vanilla/controller.go

```diff
@@ -542,8 +542,11 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("CreateVolume: called with args %+v", *req)
-
-	if common.IsFileVolumeRequest(ctx, req.GetVolumeCapabilities()) {
+	volumeCapabilities := req.GetVolumeCapabilities()
+	if !common.IsValidVolumeCapabilities(ctx, volumeCapabilities) {
+		return nil, status.Error(codes.InvalidArgument, "Volume capabilities not supported")
```

Comment:
It will be nice to return an error message saying "NFS fstype not supported for ReadWriteOnce volume creation"

To do this `IsValidVolumeCapabilities` need to tell appropriate error message.

Reply:
True, but this means we are changing the expected error message as discussed during TDS reviews:
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/tests/e2e/vsphere_file_volume_basic.go#L237
Do we still want to return error specific message ?

Reply:
Yes. For example in the case you tested where SC has fstype as "nfs" and PVC is RWO fails with "failed to provision volume with StorageClass "sc": rpc error: code = InvalidArgument desc = Volume capabilities not supported". This does not tell the proper reason that it failed since the user created a RWO volume with incorrect fsType.

Reply:
Done. Made changes to the IsValidVolumeCapabilities method to return appropriate error message. Please review the latest commit